src/tactics.ml

```diff
@@ -112,4 +112,42 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
   | P_tac_refl          ->
       handle_refine (Rewrite.reflexivity tac.pos ps)
   | P_tac_sym           ->
-      handle_refine (Rewrite.symmetry tac.pos ps)
+    handle_refine (Rewrite.symmetry tac.pos ps)
+  | P_tac_why3(s)    ->
+      (* get the goal to prove *)
+      let (hypothesis, trm) = Proof.Goal.get_type g in
+      let trm = unfold trm in
+      (* get the default or the indicated name of the prover. *)
+      let prover_name =
+        match s with
+        | None          -> Timed.(!Why3prover.current_prover)
+        | Some(name)    -> name.elt
+      in
+      (* translate from lambdapi to why3 terms. *)
+      let (l_prop, hyps, why3term) =
+          Why3prop.translate ps.proof_builtins (hypothesis, trm) in
```

Comment:
Why not directly translating `!((get_meta g).meta_type)` instead?

Reply:
Add tac.pos as argument.

Reply:
To come back to the previous remark (https://github.com/Deducteam/lambdapi/pull/171/files#r264742613): why `translate` could not take just a term as argument and no env? If you know how to translate a Prod, then you don't need to decompose your goal and have a specific translation function for handling env.

Reply:
I don't translate quantifed formulas now (equalities too), and when it will be added the translation check if the term is of the form `all a (λ (x : T a), t)` and not a `Prod` directly.

Reply:
Why not extending the translation to handle Prod(T,x.U)? If x does not occur in U (there is a function in Bindlib for this), then take trans(Prod(T,x.U)) = trans(T) => trans(U). Etc.